### PR TITLE
Initiate snapshot creation, get creation status

### PR DESCRIPTION
**Pull request by parkerduckworth**
- Initiate snapshot creation for filesystem storage
- Get snapshot creation status endpoint
- Fix conflicts between backup manager and index level snapshot creation
- Fix snapshot existence check for filesystem manager
- Regenerate swagger to add missing error payload

### CODE REVIEW

File: modules/storage-aws-s3/s3/s3.go [hunk 1]
```diff
@@ -178,6 +178,11 @@ func (s *s3) SetMetaStatus(ctx context.Context, className, snapshotID, status st
 	return nil
 }
 
+func (s *s3) InitSnapshot(ctx context.Context, className, snapshotID string) (*snapshots.Snapshot, error) {
+	// TODO: implement
```
Comment (antas-marcin) [inline modules/storage-aws-s3/s3/s3.go:182]: This should be done in a separate PR, yes?

File: modules/storage-gcs/gcs/gcs.go [hunk 2]
```diff
@@ -244,6 +244,11 @@ func (g *gcs) DestinationPath(className, snapshotID string) string {
 		makeObjectName(className, snapshotID, "snapshot.json"))
 }
 
+func (g *gcs) InitSnapshot(ctx context.Context, className, snapshotID string) (*snapshots.Snapshot, error) {
+	// TODO: implement
```
Comment (antas-marcin) [inline modules/storage-gcs/gcs/gcs.go:248]: This should be done in a separate PR, yes?

**Review by antas-marcin — APPROVED** [review]

File: modules/storage-aws-s3/s3/s3.go (hunk 1 above)
Comment (parkerduckworth) [reply modules/storage-aws-s3/s3/s3.go:182]: Yes correct :)

File: modules/storage-gcs/gcs/gcs.go (hunk 2 above)
Comment (parkerduckworth) [reply modules/storage-gcs/gcs/gcs.go:248]: Yes correct :)
